base/Dockerfile

```diff
@@ -1,55 +1,77 @@
+# If you need more, consider pulling QuantEcon/all-notebook.
 
+# Setup instructions 
+# base-notebook is the smallest one
+FROM jupyter/base-notebook 
 LABEL maintainer="QuantEcon Project <admin@quantecon.org>"
+USER root 
 
-RUN conda install pandas plotly quantecon jupyter_contrib_nbextensions && \
-pip install cufflinks
-
-ENV JULIA_PKGDIR=/opt/julia/pkg \
-    PATH=/opt/julia/bin:$PATH
-
-USER root
-
-# install julia
-RUN cd /tmp && \
-    wget --quiet -O julia_binary.tar.gz https://julialang-s3.julialang.org/bin/linux/x64/1.0/julia-1.0.0-linux-x86_64.tar.gz && \
-    rm -rf /opt/julia/src && \
-    mkdir -p /opt/julia/src && \
-    tar -C /opt/julia/src -zxf julia_binary.tar.gz --strip-components=1 && \
-    rm julia_binary.tar.gz && \
-    rm -f /opt/julia/bin/julia && \
-    mkdir -p /opt/julia/bin && \
-    ln -s /opt/julia/src/bin/julia /opt/julia/bin/julia && \
-    mkdir -p /usr/etc/julia && \
-    echo 'push!(Libdl.DL_LOAD_PATH, "$CONDA_DIR/lib")' >> /usr/etc/julia/juliarc.jl && \
-    mkdir -p $JULIA_PKGDIR && \
+# Linux setup 
+RUN apt-get update && apt-get install -yq --no-install-recommends \
+    build-essential \ 
+    git \ 
+    texlive-latex-base \
+    texlive-xetex \
+    pandoc \ 
+    libsm6 \
+    libxext-dev \
+    libxrender1 \
+    inkscape && \ 
+    apt-get clean && \ 
+    rm -rf /var/lib/apt/lists/*
+
+# Julia setup 
+# JULIA_PKGDIR will soon be deprecated; keep an eye on the jupyter upstream to see the new standard
+ENV JULIA_PKGDIR=/opt/julia
+ENV JULIA_VERSION=1.0.1
+
+# Install Julia 
+RUN mkdir /opt/julia-${JULIA_VERSION} && \
+    cd /tmp && \
+    wget -q https://julialang-s3.julialang.org/bin/linux/x64/`echo ${JULIA_VERSION} | cut -d. -f 1,2`/julia-${JULIA_VERSION}-linux-x86_64.tar.gz && \
+    tar xzf julia-${JULIA_VERSION}-linux-x86_64.tar.gz -C /opt/julia-${JULIA_VERSION} --strip-components=1 && \
+    rm /tmp/julia-${JULIA_VERSION}-linux-x86_64.tar.gz
+RUN ln -fs /opt/julia-*/bin/julia /usr/local/bin/julia
+
+# Show Julia where conda libraries are
+RUN mkdir /etc/julia && \
+    echo "push!(Libdl.DL_LOAD_PATH, \"$CONDA_DIR/lib\")" >> /etc/julia/juliarc.jl && \
+    # Create JULIA_PKGDIR \
+    mkdir $JULIA_PKGDIR && \
     chown $NB_USER $JULIA_PKGDIR && \
     fix-permissions $JULIA_PKGDIR
 
-RUN apt-get update && apt-get install curl wget && rm -rf /var/lib/apt/lists/*
+USER $NB_UID
+
+# Install Julia packages 
+# Base packages
+RUN julia -e "using Pkg; pkg\"add Expectations Compat IJulia Interpolations Revise QuantEcon InstantiateFromURL Plots GR Parameters Distributions\"; pkg\"build\"; pkg\"precompile\"" 
+# v0.3.1 QELP
+RUN julia -e "using InstantiateFromURL; activate_github(\"QuantEcon/QuantEconLecturePackages\", tag = \"v0.3.1\")"
+# Master QELP 
+RUN julia -e "using InstantiateFromURL; activate_github(\"QuantEcon/QuantEconLecturePackages\")"
+# Pkg3 garbage collection 
+RUN julia -e "using Pkg; pkg\"gc\""
```

Comment:
I think that we should combine the steps to install julia packages into one `RUN` command and make sure to have `pkg"gc"` run at the end of this command. 

As it stands right now anything that would be `gc`d by julia's Pkg is already baked into the image by the previous `RUN` commands that install packages.